async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -1095,6 +1047,28 @@ impl futures::Stream for Stream {
         if self.terminated {
             return Poll::Ready(None);
         }
+
+        if !self.batch_config.idle_heartbeat.is_zero() {
+            trace!("setting hearbeats");
+            let timeout = self.batch_config.idle_heartbeat.saturating_mul(2);
+            self.heartbeat_timeout.get_or_insert_with(|| {
+                Box::pin(tokio::time::timeout(timeout, futures::future::pending()))
```

Comment:
This could have been done with [`Sleep`](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html)

Reply:
oh, right. I took it over from different approach I had before. Nice nit. Thx.